static/app/components/superuserAccessForm.tsx

```diff
@@ -59,23 +60,24 @@ class SuperuserAccessForm extends Component<Props, State> {
     const disableU2FForSUForm = ConfigStore.get('disableU2FForSUForm');
 
     const suAccessCategory = superuserAccessCategory || data.superuserAccessCategory;
-
     const suReason = superuserReason || data.superuserReason;
 
     if (!authenticators.length && !disableU2FForSUForm) {
       this.handleError(ErrorCodes.NO_AUTHENTICATOR);
       return;
     }
 
+    // Set state to setup for U2F tap
     if (this.state.showAccessForms && !disableU2FForSUForm) {
       this.setState({
         showAccessForms: false,
         superuserAccessCategory: suAccessCategory,
         superuserReason: suReason,
       });
+    // If U2F is disabled, authenticate immediately
     } else {
       try {
-        await api.requestPromise('/auth/', {method: 'PUT', data});
+        await api.requestPromise(this.props.hasStaff ? '/staff-auth/' : '/auth/', {method: 'PUT', data});
```

Comment:
It seems like we're doing a lot of conditional logic changes based on a single parameter. Can we restructure this component to have a specific staff related component?